extensions/jpa/runtime/src/main/java/org/apache/camel/quarkus/component/jpa/DefaultTransactionStrategySubstitution.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.camel.quarkus.component.jpa;
+
+import javax.persistence.EntityManagerFactory;
+
+import com.oracle.svm.core.annotate.Substitute;
+import com.oracle.svm.core.annotate.TargetClass;
+import org.apache.camel.component.jpa.DefaultTransactionStrategy;
+import org.springframework.transaction.PlatformTransactionManager;
+import org.springframework.transaction.support.TransactionTemplate;
+
+@TargetClass(DefaultTransactionStrategy.class)
+final public class DefaultTransactionStrategySubstitution {
```

Comment:
I wonder if we could annotate this class with `@Delete` instead of doing the method substitutions? If we are forcing the transaction strategy in the extension, then in theory there should be nothing that needs access to it.

Reply:
No, graalvm has to find `DefaultTransactionStrategy` during the static analyst phase.

Reply:
It is deleted now.